eslint.config.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import pluginJs from '@eslint/js';
+import noOnlyTests from 'eslint-plugin-no-only-tests';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+  {
+    ignores: [
+      '**/*.js',
```

Comment:
I'm surprised that tseslint isn't already ignoring files not in the tsconfig. Is there a way to do that?

Otherwise, might this help? https://github.com/antfu/eslint-config-flat-gitignore

Reply:
Do you use a path argument when you run eslint in your npm script, like `eslint src/**/*.ts` or whatever?

I like to set eslint up so that you can run it with no path argument. My experience is that when you run eslint with no path argument, it will try and lint all supported files. The tseslint config may only apply to typescript project files, but eslint is still going to try and lint everything else with the non-tseslint rules.

So I think it's still good to have all your non-lintable files listed here.

Syncing with .gitignore makes sense, but I think I'm going skip that plugin since the structure changes so rarely and I don't want to add a dep. I did clean up the list, though.